packages/components/src/Combobox/Combobox.test.tsx

```diff
@@ -1,530 +1,339 @@
+    const searchInput = screen.getByPlaceholderText("Search");
 
-    fireEvent.click(button);
-    fireEvent.click(option);
+    await userEvent.click(screen.getByText(activatorLabel));
+    await userEvent.type(searchInput, "Bilbo");
+    await userEvent.click(screen.getByText("Bilbo Baggins"));
 
-    expect(onSelect).toHaveBeenCalledWith([
-      {
-        id: "1",
-        label: "Bilbo Baggins",
-      },
-    ]);
+    expect(searchInput).toHaveValue("Bilbo");
   });
 
-  it("should not clear search after making a selection", () => {
-    const { getByText, getByPlaceholderText } = render(
-      <MockMultiSelectOnSelectCombobox />,
-    );
-
-    const button = getByText("Click Me");
-    const option = getByText("Bilbo Baggins");
-    const searchInput = getByPlaceholderText("Search");
+  it("should select all options when clicking Select all", async () => {
+    renderMultiSelectCombobox();
 
-    fireEvent.click(button);
-    fireEvent.change(searchInput, { target: { value: "Bilbo" } });
-    fireEvent.click(option);
+    await userEvent.click(screen.getByText(activatorLabel));
+    await userEvent.click(screen.getByText("Select all"));
 
-    expect(searchInput).toHaveValue("Bilbo");
+    expect(handleSelect).toHaveBeenCalledWith([
+      { id: "1", label: "Bilbo Baggins" },
+      { id: "2", label: "Frodo Baggins" },
+    ]);
   });
 
   describe("onClose callback", () => {
-    it("should call onClose with selections when the content is closed", async () => {
-      const onClose = jest.fn();
-      const { getByText } = render(
-        <MockMultiSelectOnCloseCombobox onCloseOverride={onClose} />,
-      );
+    const handleClose = jest.fn();
 
-      const button = getByText("Click Me");
-      const option = getByText("Bilbo Baggins");
-      const spoder = getByText("Shelob the Spoder");
+    beforeEach(() => {
+      handleClose.mockClear();
 
-      fireEvent.click(button);
-      fireEvent.click(option);
-      fireEvent.click(spoder);
-      fireEvent.keyDown(button, { key: "Escape" });
+      render(
+        <Combobox
+          heading={activatorLabel}
+          multiSelect={true}
+          selected={[]}
+          onClose={handleClose}
+        >
+          <Combobox.Option id="1" label="Bilbo Baggins" />
+          <Combobox.Option id="2" label="Frodo Baggins" />
+          <Combobox.Action label="Add Teammate" onClick={handleAction} />
+        </Combobox>,
+      );
+    });
 
-      expect(onClose).toHaveBeenCalledWith([
+    it("should call onClose with selections when the content is closed", async () => {
+      await userEvent.click(screen.getByText(activatorLabel));
+      await userEvent.click(screen.getByText("Bilbo Baggins"));
+      await userEvent.click(screen.getByText("Frodo Baggins"));
+      await userEvent.click(screen.getByTestId(OVERLAY_TEST_ID));
+
+      // This should be 1. If this errors out and expects the value to be 1,
+      // then you've fixed the bug! Please to change the value to 1.
```

Comment:
couple things with this test wrt the big ol refactor PR

1) it won't really exist soon because `onClose` doesn't have args, it's just a lifecycle hook type thing
2) it _should_ fix this problem but yeah we can verify that once these live together